content/blog/opa-eng-plugin/opa-eng-plugin.md

```diff
@@ -0,0 +1,63 @@
+---
+title: You can use OPA for Authorization in Backstage!
+date: "2024-02-03T12:57:59Z"
+description: "Part 1: How to use OPA with Backstage for authorization"
+---
+
+Hello everyone! In this post we are going to see how you can use OPA with Backstage for authorization!
+
+## What is OPA?
+
+[OPA](https://www.openpolicyagent.org/) (pronounced “oh-pa”) is an open source, general-purpose policy engine that unifies policy enforcement across the stack. OPA is an open source project that was created by [Styra](https://www.styra.com/) and then donated to the [Cloud Native Computing Foundation](https://www.cncf.io/). It uses the declarative language Rego to define policies, more information on that [here](https://www.openpolicyagent.org/docs/latest/policy-language/).
+
+## What is Backstage?
+
+[Backstage](https://backstage.io/) is an open source framework for creating a developer portal (IDP or Internal Developer Portal). Backstage was created by Spotify and donated to the [Cloud Native Computing Foundation](https://www.cncf.io/) and has gained a lot of traction in the open source community.
+
+## Why use OPA with Backstage?
+
+Backstage uses the [Permissions Framework](https://backstage.io/docs/permissions/overview) to define permissions, this is very useful for defining permissions like RBAC in the portal, the permissions are defined in 'policies' that are written in TypeScript. This in itself is not bad and for most will be fine, but it does have some drawbacks, for example:
+
+- If you don't know TypeScript, you can't define permissions.
+- If you have a security team that is responsible for defining permissions, they have to learn TypeScript to be able to define permissions.
+- The policies can be difficult to maintain, especially if they are large.
+- You need to redeploy the code every time you change the permissions.
+
+## How can OPA help?
+
+OPA can be used to define authorization policies and then use them in Backstage with the plugin [@parsifal-m/plugin-permission-backend-module-opa-wrapper](https://www.npmjs.com/package/@parsifal-m/plugin-permission-backend-module-opa-wrapper). The plugin simply forwards the authorization requests to OPA and OPA responds with an `ALLOW` or `DENY` or `CONDITIONAL` (more on this in part 2!).
+
+This can help simplify the definition of permissions, as the permissions are defined in OPA (in Rego) and not in TypeScript, and you don't need to redeploy the code every time you change the permissions, you just need to update the policies in OPA.
+
+### What benefits does this bring?
+
+- You don't need to know TypeScript to define permissions.
+- You don't need to redeploy the code every time you change the permissions, just update the policies in OPA.
+- DevSecOps (or security teams) can define the permissions in OPA without needing to know TypeScript, (They do need to know some Rego)
```

Comment:
I would spin this as using the same tools as elsewhere, rather than not needing Typescript. Typescript might actually be familiar to people, but it's unlikely it's how permissions are defined elsewhere in the business.